lib/json/JsonParser.js

```diff
@@ -15,17 +15,20 @@ const JsonData = require("./JsonData");
 /** @typedef {import("../Module").BuildMeta} BuildMeta */
 /** @typedef {import("../Parser").ParserState} ParserState */
 /** @typedef {import("../Parser").PreparsedAst} PreparsedAst */
+/** @typedef {import("../dependencies/JsonExportsDependency").JsonDependencyOptions} JsonDependencyOptions */
 /** @typedef {import("./JsonModulesPlugin").RawJsonData} RawJsonData */
 
 const getParseJson = memoize(() => require("json-parse-even-better-errors"));
 
 class JsonParser extends Parser {
 	/**
-	 * @param {JsonModulesPluginParserOptions} options parser options
+	 * @param {JsonModulesPluginParserOptions} parserOptions parser options
+	 * @param {JsonDependencyOptions} dependencyOptions dependency options
 	 */
-	constructor(options) {
+	constructor(parserOptions, dependencyOptions) {
```

Comment:
Let's make it better and implement (in `config/defaults.js`):
```js
F(module.parser, JSON_MODULE_TYPE, () => ({}));
D(module.parser[JSON_MODULE_TYPE], "exportsDepth", mode === "development" ? 1 : Infinity);
```

So `exportsDepth` will be part of `parserOptions`, this will add some flexibility and avoid extra arguments

Reply:
Such a good idea.